### DISCUSSION TITLE
Difficulties

### DISCUSSION THREAD

**Discussion by teon101** [Cyfrin/foundry-full-course-cu#3093] [General]
hey, I'm having a hard time deploying my contract to zksync , including getting test net funds from faucets and adding zksync as a network to meta mask......can someone help me out please.thanks 
i hope someone sees this..

**Comment by EngrPips** [comment]
Hello @teon101, Have you installed Metamask browser extension on your Chrome browser?

**Comment by teon101** [comment]
Yes I have meta mask

**Comment by teon101** [comment]
But I using Firefox

**Reply by EngrPips** [reply]
I don't know if Firefox has a Metamask extension or not. Why not use Chrome instead?

**Reply by teon101** [reply]
yes , firefox has metamask extension ,that is what i have been using all this while..

**Reply by EngrPips** [reply]
You can `zksync` to your Metamask with this website https://chainlist.org/ . You can get sepolia eth from this faucet https://cloud.google.com/application/web3/faucet/ethereum/sepolia . What is the problem you are facing when try to deploy your contract to `zksync` chain?

**Comment by teon101** [comment]
1. I can't get testnet funds from any of the faucet ,after many attempts

**Reply by EngrPips** [reply]
Retry after 30 minutes this https://cloud.google.com/application/web3/faucet/ethereum/sepolia and you should get some sepolia eth.

**Reply by satyaorz** [reply]
or you can try this
https://docs.zksync.io/zksync-era/ecosystem/network-faucets
+ always check the github repo of cyfrin/foundry-full-cource for latest updated links, resources and what to use

**Reply by EngrPips** [reply]
Great heads-up.

**Comment by Ptyline** [comment]
Have you resolved your difficulty?

**Comment by teon101** [comment]
Ok I tried everything step by step , yesterday but still Got nothing...

**Reply by EngrPips** [reply]
You mean you were unable to get sepolia ether?

**Reply by teon101** [reply]
no,ZKsync

**Reply by EngrPips** [reply]
What were you unable to do regarding `ZKsync`?

**Reply by teon101** [reply]
i have sepolia ether already

**Reply by teon101** [reply]
> What were you unable to do regarding `ZKsync`? 
1. i'm enable to add zksync sapolia network
2. i'm enable to get testnet funds for zksync faucet (zksync ether)

**Reply by EngrPips** [reply]
Use this website  https://chainlist.org/ to add zksync sepolia network to your metmask. When you are done, we discuss the second problem. Also, Patrick taught how to do these things in the course so why are you having this issue? Are you not following the course?

**Reply by teon101** [reply]
ok,

**Reply by EngrPips** [reply]
Yeah.

**Reply by teon101** [reply]
ok , i have done that

**Reply by EngrPips** [reply]
Alright, To get zksync ether, the best bet is to use the zksync bridge https://portal.zksync.io/bridge to convert sepolia eth to zksync sepolia eth.

**Reply by teon101** [reply]
ok i connected my wallet but in the zksync bridge sepolia eth is not trhere

**Reply by EngrPips** [reply]
It is there 

![Screenshot 2024-12-26 at 11 18 01](https://github.com/user-attachments/assets/ab7cadaf-19f0-4223-a0fe-b2231dd2cf8a)

**Comment by teon101** [comment]
Ok got it

**Reply by EngrPips** [reply]
Good.

**Reply by teon101** [reply]
omg i sent 0.1 eth to an unkown acct

**Comment by teon101** [comment]
Ok thanks I have gotten it..so now when I deploy my contacts, will it
automatically go to zksync testnet ?

**Reply by EngrPips** [reply]
What do you mean by ` will it automatically go to zksync testnet?`

**Comment by teon101** [comment]
Sorry it was a mistake

**Comment by teon101** [comment]
Thanks, I have done it

**Comment by teon101** [comment]
Would you mind if I get to know you better 😅,maybe we can chat on a
different platform 🤔

**Reply by EngrPips** [reply]
I don't mind, Friend. Are you on Discord? Thats the social media platform I use the most.

**Reply by teon101** [reply]
ya sure i'm on discord

**Reply by EngrPips** [reply]
Are you in Cyfrin Discord?

**Comment by teon101** [comment]
Nope

**Reply by EngrPips** [reply]
Here https://discord.gg/qUEBFM2Y join Cyfrin discord

**Comment by H-code9** [comment]
Have you been able to resolve your difficulty?

**Comment by teon101** [comment]
Ya thanks for your support
